### PR TITLE
Create layout of the select seats page

### PR DESCRIPTION
**Pull request by kathrynowy**

### CODE REVIEW

File: src/Components/Hall/Hall.scss [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+  &__screen { 
+    border-top: 40px solid #3c3c3d;
+    border-left: 80px solid transparent;
+    border-right: 80px solid transparent;
+  }
+}
+
+.hall-legend__seat-type
+{
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: Leave opening bracket on the previous line

File: src/Components/Hall/Hall.scss [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    background-color: #e57373;
+  }
+}
+
+.hall-container {
+  display: flex;
+  flex-direction: column;
+  flex: 1 1 auto;
+  width: 100%;
+  .seats-info {
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: separate new class from previous styles by empty line

File: src/Components/Hall/Hall.scss [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    width: 50%;
+  }
+}
+
+.hall::-webkit-scrollbar {
+  width: 3px;
+}
+
+.hall::-webkit-scrollbar-track {
+  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: use style without box-shadow prefix

File: src/Components/Hall/Hall.scss [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    margin-top: 30px;
+    margin-bottom: 0px;
+  }
+
+  .hall__screen {
+    margin-top: 15px;
+    margin-bottom: 25px;
+    width: 450px;
+    border-top: 30px solid #3c3c3d;
+    border-left: 80px solid transparent;
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: Use padding instead

File: src/Components/Hall/Hall.scss [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+  .hall {
+    margin-top: 30px;
+    margin-bottom: 0px;
+  }
+
+  .hall__screen {
+    margin-top: 15px;
+    margin-bottom: 25px;
+    width: 450px;
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: Why it is fixed?

File: src/Components/Hall/Hall.scss [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+      margin-left: 90px;
+    }
+    
+    &__icon {
+      width: 18px;
+      height: 18px;
+    }
+
+    &__text {
+      font-size: 14px;
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: use rem for font-size

File: src/Components/Hall/Hall.scss [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+    border-left: 60px solid transparent;
+    border-right: 60px solid transparent;
+  }
+
+  .hall-legend {
+    margin-top: 20px;
+  
+    &__seat-type
+    {
+      width: 25px;
```
Comment (dzmitry-penda) [inline src/Components/Hall/Hall.scss]: Duplicate styles

File: src/Components/Hall/SeatsInfo/SeatsInfo.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import React, { Component } from 'react';
+import './SeatsInfo.scss'
+import { Link } from 'react-router-dom';
+
+
+class SeatsInfo extends Component {
+  render() {
+    const disabledBtn = this.props.selectedSeats.length ? false : true;
```
Comment (dzmitry-penda) [inline src/Components/Hall/SeatsInfo/SeatsInfo.js]: 🥇

File: src/Components/Hall/SeatsInfo/SeatsInfo.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    const newCost = this.props.selectedSeats.reduce((cost, seat) => {
```
Comment (dzmitry-penda) [inline src/Components/Hall/SeatsInfo/SeatsInfo.js]: use seat.cost instead

File: src/Components/Hall/SelectSeats/SelectSeats.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+import SeatsData from '../../../SeatsData.js'
+import './SelectSeats.scss';
+import Seat from './Seat/Seat.js'
+
+
+
+class SelectSeats extends Component {
+  handleDrawSeats = (amount, row, cost, cinemaId, movieId, hallId) => {
+    let seats = [];
+    for (let i = 0; i < amount; i++) {
```
Comment (dzmitry-penda) [inline src/Components/Hall/SelectSeats/SelectSeats.js]: i=1; i<= length; i++

File: src/Components/Hall/SelectSeats/SelectSeats.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        <Seat
+          seat={i + 1}
+          row={row}
+          cost={cost}
+          key={i + 1}
+          cinemaId={cinemaId}
+          movieId={movieId}
+          hallId={hallId}
+          isSelected={
+            this.props.selectedSeats.find((seat) => seat.id === `${row}.${i + 1}.${cost}`)
```
Comment (dzmitry-penda) [inline src/Components/Hall/SelectSeats/SelectSeats.js]: Id should not contain cost

File: src/Components/Hall/SelectSeats/SelectSeats.scss [hunk 12]
```diff
@@ -0,0 +1,230 @@
+  .row__seat_selected {
+    background-color: lightblue !important; 
+    color: black !important;
+  }
+
+  .row__number_left {
```
Comment (dzmitry-penda) [inline src/Components/Hall/SelectSeats/SelectSeats.scss]: Empty style declaration

File: src/Components/Hall/SelectSeats/SelectSeats.scss [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+    }
+  }
+}
+
+ @media only screen and (min-width: 993px) {
+  .row {
+    margin-bottom: 10px;
+
+    .row__seat {
+      margin-left: 5px;
```
Comment (dzmitry-penda) [inline src/Components/Hall/SelectSeats/SelectSeats.scss]: Duplicate styes

File: src/Components/Loginscreen/Loginscreen.js [hunk 14]
```diff
@@ -0,0 +1,46 @@
+/* import React, { Component } from 'react';
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
```
Comment (dzmitry-penda) [inline src/Components/Loginscreen/Loginscreen.js]: Unneeded file

File: src/Components/SignUp/SignUp.js [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+
+export default class SignUp extends Component {
+  render() {
+    return (
+      <div className="signup-form">
+        <form >
+
```
Comment (dzmitry-penda) [inline src/Components/SignUp/SignUp.js]: Spaces

File: src/Containers/FilmProfilePage/FilmProfilePage.js [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+        sessions={this.props.sessions}
+        cinemas={this.props.cinemas}
+      />
+    );
+  }
+}
+
+const mapStateToProps = store => ({
+  sessions: store.sessions.sessions,
+  sessionsErrored: store.sessions.sessionErrored,
```
Comment (dzmitry-penda) [inline src/Containers/FilmProfilePage/FilmProfilePage.js]: 🤔 what is it for?

File: src/actions/index.js [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+  return {
+    type: GET_SESSIONS_SUCCESS,
+    payload: sessions
+  }
+}
+
+export const getSessionsFailure = (err) => {
+  return {
+    type: GET_SESSIONS_FAILURE,
+    isErrored: err
```
Comment (dzmitry-penda) [inline src/actions/index.js]: I suppose it should be named ''error'

File: src/index.js [hunk 18]
```diff
@@ -1,8 +1,21 @@
...
+import thunk from 'redux-thunk';
+import { composeWithDevTools } from 'redux-devtools-extension';
 
+import rootReducer from "./reducers/index";
 
-ReactDOM.render(<App />, document.getElementById('root'));
 
+const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
 
+
```
Comment (dzmitry-penda) [inline src/index.js]: Spaces

File: src/Components/FilmProfile/FilmProfile.js [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+  state = {
+    day: new Date().getTime() + 1000 * 60 * 60 * 3,
+  }
+
+  getSecondsToday = () => {
+    const date = new Date();
+    return date.getHours() * 3600 + date.getMinutes() * 60 + date.getSeconds();
+  };
+
+  currentTimes = (times, day) => {
```
[the remark this answers is not in the document]
Comment (kathrynowy) [reply src/Components/FilmProfile/FilmProfile.js]: getCurrentTimes ?

File: src/Components/Hall/ConfirmTicket/TicketInfo/TicketInfo.js [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+        <div className="confirm-ticket__ticket-info">
+          <div className="confirm-ticket__number">
+            {id+1}
+          </div>
+          <div className="confirm-ticket__info">
+            <span>
+              {movieName}
+          </span>
+            <span>
+              {`row ${number.split('.')[0]}/seat ${number.split('.')[1]}`}
```
[the remark this answers is not in the document]
Comment (kathrynowy) [reply src/Components/Hall/ConfirmTicket/TicketInfo/TicketInfo.js]: i should use new variables?

[the remark this answers is not in the document]
Comment (dzmitry-penda) [reply src/Components/Hall/ConfirmTicket/TicketInfo/TicketInfo.js]: you should pass whatever you need with props

File: src/Components/FilmProfile/Schedule/Schedule.js [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class Schedule extends Component {
+  render() {
+    const { cinemaId, times, hallId, movieId, cinemas } = this.props;
+    return (
+      <div className="schedule">
+
+        <div className="schedule__cinema">{(cinemas.find(cinema => cinema.id == cinemaId)).name}</div>
+        <div className="schedule__space"></div>
```
Comment (dzmitry-penda) [inline src/Components/FilmProfile/Schedule/Schedule.js]: Padding

File: src/Components/FilmProfile/Schedule/Schedule.scss [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+      flex-grow: 1; 
+    } 
+    
+    &__space { 
+      display: flex; 
+      flex-grow: 1; 
+    } 
+    
+    &__ticket-list { 
+      flex-grow: 1;
```
Comment (dzmitry-penda) [inline src/Components/FilmProfile/Schedule/Schedule.scss]: grow or width

File: src/Components/FilmProfile/Schedule/Schedule.scss [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+    
+    &__ticket-list { 
+      justify-content: flex-start; 
+    }
+  } 
+} 
+    
+@media only screen and (max-width: 420px) { 
+  .schedule { 
+    margin-top: 15px;
```
Comment (dzmitry-penda) [inline src/Components/FilmProfile/Schedule/Schedule.scss]: Remove

File: src/Components/FilmProfile/styles.scss [hunk 24]
```diff
@@ -0,0 +1,270 @@
+.movie-profile {
+  margin-top: 55px;
+  font-family: 'Nunito','Helvetica Neue','Helvetica','Arial','sans-serif';
```
Comment (dzmitry-penda) [inline src/Components/FilmProfile/styles.scss]: Remove

File: src/Components/Hall/ConfirmTicket/TicketInfo/TicketInfo.scss [hunk 25]
```diff
@@ -0,0 +1,133 @@
...
+    right: 30px;
+    top: 50%;
+    transform: translateY(-50%);
+    cursor: pointer;
+    visibility: hidden;
+  }
+}
+
+.confirm-ticket__ticket-info {
+  width: 300;
```
Comment (dzmitry-penda) [inline src/Components/Hall/ConfirmTicket/TicketInfo/TicketInfo.scss]: 300 попугаев

File: src/Components/Hall/SeatsInfo/SeatsInfo.scss [hunk 26]
```diff
@@ -0,0 +1,144 @@
...
+    display: flex;
+    flex-direction: column;
+  }
+
+  &__btn {
+    border: 1px solid $pink; 
+    background-color: $light-brown; 
+    color: $pink; 
+    font-size: 17px; 
+    border-radius: 60px;
```
Comment (dzmitry-penda) [inline src/Components/Hall/SeatsInfo/SeatsInfo.scss]: %

File: src/Components/Hall/SeatsInfo/SeatsInfo.scss [hunk 27]
```diff
@@ -0,0 +1,144 @@
...
+    
+    &__include {
+      text-align: end;
+      align-self: flex-start;
+      width: 150px;
+    }
+  }
+}
+
+@media only screen and (min-width: 1200px) {
```
Comment (dzmitry-penda) [inline src/Components/Hall/SeatsInfo/SeatsInfo.scss]: Fix medias

File: src/Components/CardList/CardList.scss [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+.card-list {
+  justify-content: flex-start;
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+}
+
+input[type="button"]:focus {
```
Comment (dzmitry-penda) [inline src/Components/CardList/CardList.scss]: Remove unused styles

File: src/Components/CardList/CardList.scss [hunk 29]
```diff
@@ -0,0 +1,36 @@
+@import '../variables.scss';
+
+
+.container {
+  font-family: 'Nunito','Helvetica Neue','Helvetica','Arial','sans-serif';
```
Comment (dzmitry-penda) [inline src/Components/CardList/CardList.scss]: Unneeded font

File: src/Components/CardList/Card/styles.scss [hunk 30]
```diff
@@ -0,0 +1,76 @@
...
+    } 
+  }
+}
+
+@media only screen and (max-width: 768px) {
+  .card {
+    width: 140px;
+
+    &__media {
+      width: 140px;
```
Comment (dzmitry-penda) [inline src/Components/CardList/Card/styles.scss]: 100%

File: src/Components/Hall/ConfirmTicket/ConfirmTicket.scss [hunk 31]
```diff
@@ -0,0 +1,57 @@
+@import '../../variables.scss';
+
+*,
+*::before,
```
Comment (dzmitry-penda) [inline src/Components/Hall/ConfirmTicket/ConfirmTicket.scss]: nothing has changed at all

**Pull request merged**